### PR TITLE
feat: enhance OV enroll via sms flow

### PR DESCRIPTION
**Pull request by magizh-okta**
## Description:

 * Allows user to select the country code when enrolling ov via sms.
 * channel data view now has 'switch channel' link option.
* also fixed text per UX
* Bold escaped phoneNumber and email in poll view.

 RESOLVES: OKTA-326108

For new changes 
Screen Shot 2020-09-03 at 10 28 00 AM
Screen Shot 2020-09-03 at 10 30 20 AM
Screen Shot 2020-09-03 at 11 31 35 AM

Screen Shot 2020-09-02 at 2 30 10 PM
ov-enroll-sms

**Phone view still looks same after my changes** 
Screen Shot 2020-09-02 at 3 04 39 PM

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-326108](https://oktainc.atlassian.net/browse/OKTA-326108)

### CODE REVIEW

**Review by stevennguyen-okta — CHANGES_REQUESTED** [review]
Overall looks good, one minor nit: In the "sms sent" screen, can you add emphasis to the user's email? I think we already have an existing styling for it, but it looks semibolded

File: assets/sass/v2/_enroll-phone.scss [hunk 1]
```diff
@@ -2,11 +2,6 @@
   .okta-form-subtitle {
     text-align: center;
   }
-  .chzn-container {
-    // TODO: Find a way around important
-    /* stylelint-disable declaration-no-important */
-    width: 100% !important;
-  }
```
Comment (anipendakur-okta) [inline assets/sass/v2/_enroll-phone.scss:9]: Looks like this is coming from '.o-form-wide' now. Please update the PR with screenshots for Voice Enrollment as well.

Comment (magizh-okta) [reply assets/sass/v2/_enroll-phone.scss:9]: yeah it is an option in courage inputs. 

Screen Shot 2020-09-02 at 9 18 54 PM

File: playground/mocks/data/idp/idx/authenticator-enroll-ov-via-email.json [hunk 2]
```diff
@@ -156,7 +156,8 @@
                       },
                       {
                         "name": "channel",
-                        "required": false,
+                        "required": true,
+                        "visible": true,
```
Comment (gowthamidommety-okta) [inline playground/mocks/data/idp/idx/authenticator-enroll-ov-via-email.json:160]: Why is this visible: true now?

File: src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js [hunk 3]
```diff
@@ -1,29 +1,127 @@
...
+        },
+        Object.assign({}, phoneNumberUISchema),
+      ],
+    };
+
+    if (phoneNumberUISchemaIndex !== -1) {
+      // Replace phoneNumberUISchema..
+      uiSchemas.splice(phoneNumberUISchemaIndex, 1, phoneNumberWithCodeUISchema);
+      // Add countryUISchema before phone.
+      uiSchemas.splice(phoneNumberUISchemaIndex, 0, countryUISchema);
```
Comment (gowthamidommety-okta) [inline src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js:52]: splice can take multiple items to add right? Could we not do this in one step?

File: src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js [hunk 4]
```diff
@@ -1,29 +1,127 @@
...
+
+  handlePhoneCodeChange () {
+    const countryCodeField = this.el.querySelector('.country-code-label');
+    countryCodeField.innerText = `+${this.model.get('phoneCode')}`;
+  },
+
+  initialize () {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    if (this.options.appState.get('currentAuthenticator').contextualData.selectedChannel === 'sms') {
+      this.listenTo(this.model, 'change:phoneCode', this.handlePhoneCodeChange.bind(this));
```
Comment (gowthamidommety-okta) [inline src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js:74]: Is there a modelEvents we could put this logic?

File: playground/mocks/data/idp/idx/authenticator-enroll-ov-via-email.json (hunk 2 above)
Comment (magizh-okta) [reply playground/mocks/data/idp/idx/authenticator-enroll-ov-via-email.json:160]: Channel selection needs to be visible. I have it as visible everywhere else and just dint update this json earlier.

File: playground/mocks/data/idp/idx/authenticator-enroll-ov-via-sms.json [hunk 5]
```diff
@@ -48,27 +48,30 @@
...
-                                  "label":"QRCODE"
-                              },
-                              {
-                                  "value":"sms",
-                                  "label":"SMS"
-                              }
-                            ]
+                          "name":"channel",
+                          "required":true,
+                          "visible": true,
```
[the remark this answers is not in the document]
Comment (magizh-okta) [reply playground/mocks/data/idp/idx/authenticator-enroll-ov-via-sms.json:60]: yeah. Backend it doesnt matter  as long as we send a value and in this case it expects a channel.

File: src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js (hunk 4 above)
Comment (magizh-okta) [reply src/v2/view-builder/views/ov/EnrollementChannelDataOktaVerifyView.js:74]: We can but I wanted to selectively add the listener so I Added it here.

File: src/v2/view-builder/views/ov/EnrollChannelPollDescriptionView.js [hunk 6]
```diff
@@ -15,13 +15,13 @@ export default View.extend({
       {{/if}}
       {{#if email}}
         <ul class="email-info ov-info">
-          <li>{{i18n code="oie.enroll.okta_verify.email" bundle="login" arguments="email"}}</li>
+          <li>{{{i18n code="oie.enroll.okta_verify.email.info" bundle="login" arguments="email"}}}</li>
```
Comment (gowthamidommety-okta) [inline src/v2/view-builder/views/ov/EnrollChannelPollDescriptionView.js:18]: Do we need to add an explain comment here?

**Review by haishengwu-okta — COMMENTED** [review]
LGTM

**Review by gowthamidommety-okta — APPROVED** [review]
lgtm after comment above 🚢

File: src/v2/view-builder/views/ov/EnrollChannelPollDescriptionView.js (hunk 6 above)
Comment (magizh-okta) [reply src/v2/view-builder/views/ov/EnrollChannelPollDescriptionView.js:18]: We do this in many places and also we are explicitly escaping the phone I dont see a need for comment.

**Pull request closed without merging**